pkg/controller/kubelet-config/kubelet_config_infrastructure.go

```diff
@@ -0,0 +1,304 @@
+		glog.V(2).Infof(message)
+		ctrl.eventRecorder.Eventf(infraConfig, corev1.EventTypeNormal, "ActionProhibited", message)
+		return
+	}
+	glog.V(4).Infof("Adding the infrastructure config resource, name: %s", infraConfig.Name)
+	ctrl.enqueueInfraConfig(infraConfig)
+}
+
+func (ctrl *Controller) deleteInfraConfig(obj interface{}) {
+	infraConfig, ok := obj.(*osev1.Infrastructure)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("couldn't retrieve the object from the Add Infrastructure Config event %#v", obj))
+		return
+	}
+	if infraConfig.Name != ctrlcommon.ClusterInfrastructureInstanceName && infraConfig.Status.CPUPartitioning == osev1.CPUPartitioningAllNodes {
+		glog.V(4).Infof("Infrastructure config %s was deleted, turning off CPUPinning is not supported, no action will be taken", infraConfig.Name)
+	}
+}
+
+func RunInfraConfigBootstrap(infraConfig *osev1.Infrastructure, mcpPools []*mcfgv1.MachineConfigPool) ([]*mcfgv1.MachineConfig, error) {
+	configs := []*mcfgv1.MachineConfig{}
+
+	if infraConfig == nil {
+		return configs, fmt.Errorf("infrastructure resource is nil, can not generate CPU Partitioning")
+	}
+
+	if infraConfig.Status.CPUPartitioning != osev1.CPUPartitioningAllNodes {
+		return configs, nil
+	}
+
+	for _, pool := range mcpPools {
+		role := pool.Name
+
+		// Get MachineConfig
+		mc, err := cpuPartitionMC(role, cpuPartitionIgn())
+		if err != nil {
+			return nil, err
+		}
+		configs = append(configs, mc)
+	}
+
+	return configs, nil
+}
+
+func cpuPartitionMC(role string, ignitionConfig ign3types.Config) (*mcfgv1.MachineConfig, error) {
+	rawIgnition, err := json.Marshal(ignitionConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	mc := &mcfgv1.MachineConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: mcfgv1.GroupVersion.String(),
+			Kind:       "MachineConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cpuPartitionMCName(role),
+			Labels: map[string]string{
+				"machineconfiguration.openshift.io/role": role,
+			},
+			Annotations: map[string]string{
+				ctrlcommon.GeneratedByControllerVersionAnnotationKey: version.Hash,
+			},
+		},
+		Spec: mcfgv1.MachineConfigSpec{},
+	}
+	mc.Spec.Config = runtime.RawExtension{Raw: rawIgnition}
+	return mc, nil
+}
+
+func cpuPartitionIgn() ign3types.Config {
+	mode := 420
+	source := "data:text/plain;charset=utf-8;base64,ewogICJtYW5hZ2VtZW50IjogewogICAgImNwdXNldCI6ICIiCiAgfQp9Cg=="
+	return ign3types.Config{
+		Ignition: ign3types.Ignition{
+			Version: ign3types.MaxVersion.String(),
+		},
+		Storage: ign3types.Storage{
+			Files: []ign3types.File{{
+				Node: ign3types.Node{
+					Path: "/etc/kubernetes/openshift-workload-pinning",
+				},
+				FileEmbedded1: ign3types.FileEmbedded1{
+					Contents: ign3types.Resource{
+						Source: &source,
+					},
+					Mode: &mode,
+				},
+			}},
+		},
+	}
+}
+
+func cpuPartitionMCName(role string) string {
+	return fmt.Sprintf("01-%s-cpu-partitioning", role)
```

Comment:
So, this has a `01` prefix which means it is easily overwriteable by other MCs, which should be fine, since if I understand correctly, `/etc/kubernetes/openshift-workload-pinning` is the only file this will touch? And nothing else should be touching that file today?

Reply:
Actually, that file will be overwritten by other MCs later on. The intention is to set that file up originally to configure workload partitioning on the kubelet, then afterwords, when the user configures the PerformanceProfile via NTO, that file will be overwritten with CPUSet information that is defined in the PerformanceProfile. NTO is the one that will generate MCs to control that file afterwords.